### PR TITLE
os/bluestore: some cleanup/refactor around onode in-memory representation

### PR DESCRIPTION
**Pull request by ifed01**
Signed-off-by: Igor Fedotov <ifedotov@mirantis.com>

### CODE REVIEW

File: src/os/bluestore/BlueStore.h [hunk 1]
```diff
@@ -212,27 +212,20 @@ class BlueStore : public ObjectStore,
...
     // it uses less memory and we expect this to be very small (very
     // few IOs in flight to the same Blob at the same time).
     state_list_t writing;   ///< writing buffers, sorted by seq, ascending
 
-    BufferSpace(Cache *c) : cache(c) {
-      if (cache) {
-	cache->add_blob();
-      }
+    BufferSpace(){
     }
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.h]: @liewegas - removed.

**Comment by liewegas** [comment]
needs rebase

**Comment by ifed01** [comment]
@liewegas - rebased

**Comment by liewegas** [comment]
fails bluestore unit tests...

**Comment by ifed01** [comment]
retest this please

**Comment by ifed01** [comment]
retest this please

**Comment by ifed01** [comment]
Closing in favor of PR #12700

**Pull request closed without merging**